tests/core/test_variable.py

```diff
@@ -264,6 +264,12 @@ def test_init(self):
         self.assertDictEqual({p:v for p,v in w.depends}, {'t':t,'u':u,'v':v})
         self.assertAlmostEqual(w.value, 6.0)
 
+        #test creation with scalar attributes
+        w = DepVar(t=1.0, u=2.0, v=3.0)
+        self.assertCountEqual(w.params, ('t','u','v'))
+        self.assertDictEqual({p:v.value for p,v in w.depends}, {'t':1.0,'u':2.0,'v':3.0})
+        self.assertAlmostEqual(w.value, 6.0)
+
```

Comment:
Can you try a test where you set one of these attributes (`w.t = 4.0`) as a scalar after construction?